Source/WebCore/platform/graphics/ImmersiveVideoMetadata.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <WebCore/DoubleSize.h>
+#include <WebCore/IntSize.h>
+#include <optional>
+#include <wtf/JSONValues.h>
+#include <wtf/RefPtr.h>
+
+#if PLATFORM(COCOA)
+#include <simd/matrix.h>
+#endif
+
+namespace WebCore {
+
+enum class VideoProjectionMetadataKind : uint8_t {
+    Unknown,
+    Rectilinear,
+    Equirectangular,
+    HalfEquirectangular,
+    EquiAngularCubemap,
+    Parametric,
+    Pyramid,
+    AppleImmersiveVideo,
+};
+
+enum class HeroEye : uint8_t {
+    Left,
+    Right,
+};
+
+enum class ViewPackingKind : uint8_t {
+    SideBySide,
+    OverUnder,
+};
+
+enum class LensAlgorithmKind : uint8_t {
+    ParametricLens,
+};
+
+enum class LensDomain : uint8_t {
+    Color,
+};
+
+enum class LensRole : uint8_t {
+    Mono,
+    Left,
+    Right,
+};
+
+enum class ExtrinsicOriginSource : uint8_t {
+    StereoCameraSystemBaseline,
+};
+
+using RadialDistortionCoefficients = Vector<float, 4>;
+using LensFrameAdjustmentsPolynomial = Vector<float, 3>;
+using ExtrinsicOrientationQuaternion = Vector<float, 3>;
```

Comment:
Are these really `Vector`s, as in resizable arrays?
Otherwise if they're always fixed-size, consider using std::array instead.